integrationTests/testInitializer.go

```diff
@@ -622,3 +647,497 @@ func extractUint64ValueFromTxHandler(txHandler data.TransactionHandler) uint64 {
+		Data:     args.data,
+	}
+	txBuff, _ := TestMarshalizer.Marshal(tx)
+	tx.Signature, _ = signer.Sign(skSign, txBuff)
+
+	return tx
+}
+
+func skToPk(sk crypto.PrivateKey) []byte {
+	pkBuff, _ := sk.GeneratePublic().ToByteArray()
+	return pkBuff
+}
+
+// TestPrivateKeyHasBalance checks if the private key has the expected balance
+func TestPrivateKeyHasBalance(t *testing.T, n *TestProcessorNode, sk crypto.PrivateKey, expectedBalance *big.Int) {
+	pkBuff, _ := sk.GeneratePublic().ToByteArray()
+	addr, _ := TestAddressConverter.CreateAddressFromPublicKeyBytes(pkBuff)
+	account, _ := n.AccntState.GetExistingAccount(addr)
+	assert.Equal(t, expectedBalance, account.(*state.Account).Balance)
+}
+
+// GetMiniBlocksHashesFromShardIds returns miniblock hashes from body
+func GetMiniBlocksHashesFromShardIds(body dataBlock.Body, shardIds ...uint32) [][]byte {
+	hashes := make([][]byte, 0)
+
+	for _, miniblock := range body {
+		for _, shardId := range shardIds {
+			if miniblock.ReceiverShardID == shardId {
+				buff, _ := TestMarshalizer.Marshal(miniblock)
+				hashes = append(hashes, TestHasher.Compute(string(buff)))
+			}
+		}
+	}
+
+	return hashes
+}
+
+// EqualSlices checks if two slices are equal
+func EqualSlices(slice1 [][]byte, slice2 [][]byte) bool {
+	if len(slice1) != len(slice2) {
+		return false
+	}
+
+	//check slice1 has all elements in slice2
+	for _, buff1 := range slice1 {
+		found := false
+		for _, buff2 := range slice2 {
+			if bytes.Equal(buff1, buff2) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+
+	//check slice2 has all elements in slice1
+	for _, buff2 := range slice2 {
+		found := false
+		for _, buff1 := range slice1 {
+			if bytes.Equal(buff1, buff2) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+	return true
+}
+
+// GeneratePrivateKeyInShardId generates a private key that is in a given shard
+func GeneratePrivateKeyInShardId(
+	coordinator sharding.Coordinator,
+	shardId uint32,
+) crypto.PrivateKey {
+
+	suite := kyber.NewBlakeSHA256Ed25519()
+	keyGen := signing.NewKeyGenerator(suite)
+	sk, pk := keyGen.GeneratePair()
+
+	for {
+		buff, _ := pk.ToByteArray()
+		addr, _ := TestAddressConverter.CreateAddressFromPublicKeyBytes(buff)
+
+		if coordinator.ComputeId(addr) == shardId {
+			return sk
+		}
+
+		sk, pk = keyGen.GeneratePair()
+	}
+}
+
+// GenerateRandomHexAddressInShard generates a random address in the given shard
+func GenerateRandomHexAddressInShard(
+	coordinator sharding.Coordinator,
+	addrConv state.AddressConverter,
+) string {
+
+	addrBytes := []byte(CreateRandomHexString(32))
+	for {
+		addr, _ := addrConv.CreateAddressFromPublicKeyBytes(addrBytes)
+		if coordinator.ComputeId(addr) == coordinator.SelfId() {
+			return hex.EncodeToString(addrBytes)
+		}
+		addrBytes = []byte(CreateRandomHexString(32))
+	}
+}
+
+// CreateMintingForSenders creates account with balances for every node in a given shard
+func CreateMintingForSenders(
```

Comment:
Maybe use MintAllNodes or MintAddress?

Reply:
Can't, because it needs to mint only for sendersPrivateKeys.